ruby/ext/google/protobuf_c/repeated_field.c

```diff
@@ -47,6 +47,15 @@ RepeatedField* ruby_to_RepeatedField(VALUE _self) {
   return self;
 }
 
+int index_position(VALUE _index, RepeatedField* repeated_field){
+  int index = NUM2INT(_index);
+  if (index < 0 && repeated_field->size > 0) {
+    index = repeated_field->size + index;
+  }
+  return index;
```

Comment:
Yep, file-internal helpers don't need to go in the header. Probably good to add a `static` though (this keeps the symbol internal to the file, like an anonymous namespace in C++).